### DISCUSSION TITLE
Feature Request: regex_span and regex_span_all

### DISCUSSION THREAD

**Discussion by wasade** [duckdb/duckdb#13154] [Ideas]
In Python, using the `re` module, it is easy to obtain the start and stop coordinates of a match. It would be pleasant to have similar functionality in DuckDB. My impression is that `RE2` supports obtaining this. 

```python
>>> re.compile(r"foo").search('asdasdfooasewrwbarasdefooerwfdgdf')
<re.Match object; span=(6, 9), match='foo'>
>>> re.compile(r"foo").search('asdasdfooasewrwbarasdefooerwfdgdf').span()
(6, 9)
>>> scan = re.compile(r"foo").scanner('asdasdfooasewrwbarasdefooerwfdgdf')
>>> scan.search()
<re.Match object; span=(6, 9), match='foo'>
>>> scan.search()
<re.Match object; span=(22, 25), match='foo'>
```

**Comment by franco-avi** [comment]
It would be nice to have, currently you can only do something like ```position(regexp_extract(text, 'foo.*') in text)``` for obtaining the start position.
